### DISCUSSION TITLE
setupListOperation

### DISCUSSION THREAD

**Discussion by blondie63** [Laravel-Backpack/community-forum#452] [Q&A (Help)]
I've this code:
```php
    protected function setupCreateOperation()
    {
        CRUD::setValidation(EmailTemplateRequest::class);

        CRUD::field('company');
        CRUD::field('template_type')->type('select_from_array')->options(['1' => 'booking_reject', '2' => 'booking_confirmed', '3' => 'verify_email']);
        CRUD::field('language')->type('select_from_array')->options(['it' => 'it', 'en' => 'en']);
        CRUD::field('content')->type('tinymce');
```

and i need to create a custom setupListOperation to see options instead code for field 'template_type'

Some help?
Thanks

**Comment by jcastroa87** [comment]
Hi @blondie63 i hope you are doing well, could you provide us an image or more detail example to understand what do you need so we can see if thats is possible using Backpack.

Thanks

**Comment by blondie63** [comment]
Hi @jorgetwgroup , thanks for your reply
It's very simple ! try to look these screenshots

<img width="1162" alt="ASH-Booking-et-array" src="https://user-images.githubusercontent.com/1261851/227536320-356a8870-008a-4f3b-a92f-0b7f4dfc1cde.png">
<img width="1350" alt="ASH-Booking-et-array2" src="https://user-images.githubusercontent.com/1261851/227536330-8137d3ce-bd57-4aa9-9a61-828bdaea3392.png">

I can do easily use a db table and relationship but i don't know how to do using an array like code on my first post..

**Reply by jcastroa87** [reply]
Got it, in your "setupListOperation" define the column like this

```
[   // select_from_array
    'name'    => 'template_type',
    'label'   => 'template_type',
    'type'    => 'select_from_array',
    'options' => ['1' => 'booking_reject', '2' => 'booking_confirmed', '3' => 'verify_email'],
],
```

To this become perfect, maybe you can define the array in your model and use it wherever you want.

Let my know if you need any other stuff.

**Comment by blondie63** [comment]
@jorgetwgroup Thanks, it works fine !
